include/oneapi/dpl/internal/function.h

```diff
@@ -85,7 +85,12 @@ get_access(Policy, counting_iterator<T> i)
 
 template <sycl::access::mode Mode, typename Policy, typename T>
 T*
-get_access(const Policy& policy, T* ptr)
+get_access(const Policy&
```

Comment:
May be better to do something like this ?
https://stackoverflow.com/questions/1486904/how-do-i-best-silence-a-warning-about-unused-variables
My opinion - new code state is not good readable.

Reply:
@SergeyKopienko, do you mind if we continue discussion about the way we suppress unused parameter warnings here?
You have written below:
> My proposal - required some define.
For example, PARAM_USE or something other.
> In this case you may use this define in all your changes from this commit.

I believe we can leave the current state: omit parameters where possible and use a cast to void in the special case with an assert. It is simpler than adding an extra define. Moreover, there are many other places, where we omit parameters like it is done in this patch: see #82.

Reply:
I discussed this with @SergeyKopienko offline and he suggested using the macro instead of omitting parameters due to easier debugging: if we use the macro we can analyze the parameter (e.g. to check its state).

Reply:
@SergeyKopienko I applied an approach similar to the suggested one, but without macros. Please, check `__maybe_unused` function.